Update DAOs

jeffreyw2025 opened the pull request:
This updates the PlayerDAO and GameDAO classes to reflect the changes to the database. 

- Update methods were introduced to GameDAO (aggregate update) and PlayerDAO (for cards). 
- Implemented Comparable for Players and changed arrays to ArrayLists so that we can easily sort players (for updating turn order and final payout).
- Miscellaneous fixes.

MadTurt1e committed:
A standard turn has been added with everything that goes on there.

MadTurt1e committed:
Updated files, fixed errors.
Luigi class exists - AI to be completed.

MadTurt1e committed:
Fixed more errors for the game

MadTurt1e committed:
Merge branch 'main' into Game_Turn

MadTurt1e committed:
updated gitignore with more files.

MadTurt1e committed:
Noticed error relating to connection and their try/catch blocks

jeffreyw2025 committed:
Starting to refactor DAOs

MadTurt1e committed:
Merge branch 'Game_implementation' into Game_Turn

# Conflicts:
#	.idea/workspace.xml
#	src/main/java/game/jdbc/picturepoker/PlayerDAO.java

MadTurt1e committed:
Updated PlayerDAO again to not be bad. I don't see PlayerDAO errors, on this side, however.

MadTurt1e committed:
Reverted Game database to not be broken anymore - we must keep 4 players in the database.

MadTurt1e committed:
Reverted back to original game table

jeffreyw2025 committed:
Updated DAOs, implemented player comparison

jeffreyw2025 committed:
Added hand functionality to PlayerDAO, Miscellaneous fixes

MadTurt1e commented on src/main/java/game/jdbc/picturepoker/GameDAO.java:
```diff
@@ -45,6 +40,23 @@ public Game findById(long id){
         return game;
     }
 
+    public ArrayList<Long> getPIDsByGame(Game dto){
```
Thank you for making a getPID function
I will probably update all of my PID get functions accordingly.

MadTurt1e commented on src/main/java/game/jdbc/picturepoker/GamePlay.java:
```diff
@@ -0,0 +1,110 @@
+    private boolean executeLuigi() {
+        //I believe Luigi should be his own class.
+        // The class will be referenced here.
+        Luigi luigi = new Luigi();
```
I agree that Luigi probably can just be set as a player instead which will be easier. Also will be changed eventually.

MadTurt1e commented on src/main/java/game/jdbc/picturepoker/PicturePokerGame.java:
```diff
@@ -1,7 +1,7 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
-import jdk.internal.misc.InnocuousThread;
+// import jdk.internal.misc.InnocuousThread; // Why is this here?
```
This is from the sample code provided in class - I also am not too certain what it does.

MadTurt1e commented on src/main/java/game/jdbc/picturepoker/GameDAO.java:
```diff
@@ -91,4 +103,23 @@ public Game update_string(String attribute, String data, Game dto){
         }
     }
+
+    // This is the second time I had to program this. Richard, please don't go around reverting my changes willy-nilly.
```
Sorry about this. 
Github features do include looking at past code, so if you ever want to revert code (revert reverts, or something like that), you can just re access previous commits, and physically copy and paste the text (if you don't want to deal with git revert, and all that messy stuff). 

Sorry about the mess up, hopefully this doesn't happen again.

MadTurt1e commented on .gitignore:
```diff
@@ -44,3 +44,15 @@ build/
 ### VS Code ###
 .vscode/
 
+# individual idea files
```
You can add "personal files" that change computer to computer into the gitignore if you want - like that xml file, which keeps on changing every merge. 

I predict that if we start using vscode more, we may need to add more things to the Gitignore.

MadTurt1e commented on src/main/java/game/jdbc/picturepoker/PlayerDAO.java:
```diff
@@ -11,9 +11,11 @@ public class PlayerDAO extends DataAccessObject<Player>{
     + "fourth_places, lifetime_tokens, tokens, bet, rounds_won FROM player WHERE p_id = ?";
-    private static final String CREATE_NEW_PLAYER = "INSERT INTO player (p_name, password) VALUES (?, ?)";
+    private static final String CREATE_NEW_PLAYER = "INSERT INTO player (p_name, passcode) VALUES (?, ?)";
+    private static final String CREATE_NEW_CARD = "INSERT INTO card (pid, hand_pos, suit) VALUES (?, ?, ?)";
```
pid may be wrong - you may have meant to write p_id (note that I don't know your new database layout: this could be a completely intentional spelling of p_id)

MadTurt1e replied:
Also, I just noticed that card should probably be renamed player_card (as the table has been renamed). Again, I'm not as familiar with the new database - this could be completely correct structure.

MadTurt1e reviewed (commented):
Good job on all of the changes. The database is looking pretty good, and everything is looking to be working quite nicely. (apart from that one [possible] typo)
I'll work on updating the actual game functions with these fancy new functions.

MadTurt1e committed:
Fixed a few typos within the sql code - python file should create a good database.

Also, uncomment the line DROP DATABASE picturepoker; at the beginning of databaseCreation.sql if you wish to refresh your databases.

MadTurt1e reviewed (changes requested):
Also noticed potential typo with card vs player card - once the two typos(?) are fixed (or confirmation that this isn't a typo), we should be good to merge.

jeffreyw2025 committed:
Fixed tpyos (that's totally how it's spelled) and added dealer_card functionality

jeffreyw2025 committed:
Implemented dealer_card table

MadTurt1e reviewed (approved):
Good job!
